source/extensions/filters/http/ratelimit/ratelimit.cc

```diff
@@ -267,8 +293,12 @@ void Filter::populateRateLimitDescriptors(const Router::RateLimitPolicy& rate_li
             fmt::format("ratelimit.{}.http_filter_enabled", disable_key), 100)) {
       continue;
     }
-    rate_limit.populateDescriptors(descriptors, config_->localInfo().clusterName(), headers,
-                                   callbacks_->streamInfo());
+    const bool apply_on_stream_done = rate_limit.applyOnStreamDone();
+    has_apply_on_stream_done_policy_ |= apply_on_stream_done;
```

Comment:
I will try to let this function state less and try to do a quick iteration at the stream is done. We won't have two much policies for the route/vhost, so, this wont bring too much overhead.

Reply:
done